src/engine/v3/V3Interpreter.v3

```diff
@@ -15,12 +15,11 @@ class V3Interpreter extends WasmStack {
 	private var params_arity = -1;
 	private var return_arity = -1;
 
-	// After `resume` is called, this stack is marked as CALL_CHILD and moves to
-	// the next instruction when resumed in the future. However at that time,
-	// the `pc` would point to the next instruction, making handler lookup
-	// impossible without somehow retrieving the `pc` at the `resume` that
-	// caused the CALL_CHILD transition. This variable is a hack for that.
-	private var prev_resume_pc = -1;
+	// XXX: remove this
+	// This flag is used for incrementing past a {resume} or {switch} opcode after
+	// being resumed. Suspending/switching shouldn't increment {frame.pc} until
+	// {resume} as {resume_throw} uses the old {pc} for error handling.
```

Comment:
Is it possible to check for all errors before incrementing the PC? IIUC after suspending, the stack is left in a resumable state with the `pc` at the next instruction. Is it true the only error that can happen after calling the runtime to search for a handler that no handler is found? Then it seems that should be possible to handle in the caller and either update the `pc` before switching to the new stack, or trapping.

Reply:
Sorry I don’t really understand. I was thinking that after a stack suspends, its pc points to the next instruction being transferring to the parent stack. Therefore if resume_throw is called on this suspended stack, the throw pc will be misplaced (the expected throw pc of a resume_throw should be the pc of the suspend/switching instruction that suspended the stack). Are you suggesting that this adjustment should be done by the parent that called resume_throw?

Reply:
Ah, right, hadn't thought about that. What about storing the suspending `pc` separately? That's maybe a similar amount of complexity but less subtle. It shouldn't make any difference for finding a handler in `resume_throw` because before or after the suspend will be in the same control scope (and thus have the same handlers).

Reply:
I think this differs from the reference interpreter behavior (e.g., on [test/regress/ext:stack-switching/resume_throw9.wast](test/regress/ext:stack-switching/resume_throw9.wast), where the tag thrown by `resume_throw` is supposed to be caught by the `try_table`).

Reply:
AFAICT we could fix that by considering the `end` of that `try_table` to be inside the `try_table`. We should also develop regular EH tests for such boundary cases.